app/src/main/java/lt/vilnius/tvarkau/NewProblemActivity.java

```diff
@@ -230,8 +231,16 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                     break;
                 case REQUEST_PLACE_PICKER:
                     Place place = PlacePicker.getPlace(this, data);
-                    mAddProblemLocation.setText(place.getName());
-                    locationCords = place.getLatLng();
+                    CharSequence address = place.getAddress();
+                    String matchVilnius = "Vilnius";
+                    Boolean isVilnius = Arrays.asList(address.toString().split(" ")).contains(matchVilnius);
```

Comment:
This seems really hacky. Do we have any guarantees about address format ? 
Consider such example: `Gabijos 47-7 Vilnius, Lietuva`. Such address won't work.
I would recommend to check place coordinates if they are inside Vilnius boundary instead of depending on address string.